src/nvim/os/env.c

```diff
@@ -169,6 +178,64 @@ void os_get_hostname(char *hostname, size_t size)
 #endif
 }
 
+/// @return -1 if error encountered, 1 if capslock active otherwise 0 if not.
+int capslock_status(void ) {
```

Comment:
This hardly fits into `env.c`, `input.c` would be a bit better.

Reply:
@ZyX-I I suggested env.c because input.c thus far has avoided much cross-platform junk.

Reply:
It is still out of theme. Better add `keyboard.c` then turn `env.c` into yet another incarnation of all-including `util.c`.

Reply:
OK, I will edit it